### PR TITLE
Introduce A Few More Volume Sanity Checks

### PR DESCRIPTION
**Pull request by irishgordo**
- introduce few more baseline coverages of volume functionality
- reduce network chatter, avoid making a call to detach if
volume doesn't have a linode_id attached to it
- tox.ini, update for localhost, helps speed being able to drop a
debug breakpoint within a given test and work with tox/pytest
- update client test to account for additional fixture
- introduce additional volume fixture to represent attached
volume from the start to better isolate detach functionality
check

Resolves: tests/volume-sanity-updates

### CODE REVIEW

**Comment by irishgordo** [comment]
Hey there!
Hope all is well!
This API is rad!
Happened to see a possible opportunity to increase some baseline sorta sanity coverage on Volumes and thought it might be worth a PR.
Super open to iterate, make adjs, take input, or what-have-ya!

**Comment by Dorthu** [comment]
Thanks for the contribution!

Generally, this library doesn't do a lot of validation since it isn't the authority on the current state of objects.  In this case, it would be possible for me to write code like this: 

```python
from linode_api4 import LinodeClient, Volume

client = LinodeClient(TOKEN)

my_volume = client.load(Volume, VOLUME_ID)
# my_volume.linode_id is None

# perform some time-consuming operation
# during this time, my_volume may end up being attached to a Linode through some
# other process.  However, the local object `my_volume` doesn't know this.

my_volume.detach() # with a client-side check, this would be. a no-op even though my_volume is now attached
```

We could avoid the above by reloading the Volume object periodically, or before each use, but that wouldn't reduce the number of requests being made to the API, and in some cases would result in more.

I would generally prefer this library to allow the API to determine if operations can succeed, even if it means the occasional
unneeded API call.  If the extra call is an issue, would you be opposed to making this behavior optional, either through a new `detach_if_not_attached` method, or with a `if_attached` keyword argument that defaults to using the current behavior?

**Comment by irishgordo** [comment]
Hey @Dorthu  thanks for the quick response!

Things definitely click a bit more for me now!  I did notice that the library doesn't do a ton of validation but now it makes more sense as to why :).  Yeah, I wouldn't see there being an issue with the extra API call at all :) in this case.

I've gone ahead and made some edits to revert/re-implement the default behavior of detach, remove a duplicate sanity-stylized-kinda test and also removed some opinionated tox.ini edits.

**Review by Dorthu — APPROVED** [review]
